.github/workflows/mergability_check.yml

```diff
@@ -0,0 +1,41 @@
+name: Validate and check if PR is mergable
+
+on:
+  pull_request:
+
+jobs:
+  do_merge_dry_run:
+    name: mergability_check_${{ github.event.pull_request.number }}
+    runs-on: linux.20_04.4x
+    env:
+        GH_RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+    steps:
+      - name: Checkout repo
+        id: checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
fetch-depth: 0 means " all history for all branches and tags".
Do we need this?